dev-libs/dietlibc/files/dietlibc-0.34-respect-cc-env.patch

```diff
@@ -0,0 +1,84 @@
+diff --git a/Makefile b/Makefile
+index a4a7a2b..7ca733c 100644
+--- a/Makefile
++++ b/Makefile
+@@ -117,10 +117,10 @@ all: $(WHAT)
+ profiling: $(OBJDIR)/libgmon.a $(OBJDIR)/pstart.o
+ 
+ DEFAULTCFLAGS=-pipe -nostdinc -D_REENTRANT $(EXTRACFLAGS)
+-CFLAGS=$(DEFAULTCFLAGS)
++CFLAGS += $(DEFAULTCFLAGS)
+ CROSS=
+ 
+-CC=gcc
++CC?=gcc
+ CCC=$(CROSS)$(CC)
+ STRIP=$(COMMENT) $(CROSS)strip
+ INC=-I. -isystem include
+@@ -157,7 +157,7 @@ CFLAGS+=-O -fomit-frame-pointer
+ endif
+ 
+ ifneq ($(DEBUG),)
+-CFLAGS = -g $(EXTRACFLAGS)
++CFLAGS ?= -g $(EXTRACFLAGS)
+ STRIP = :
+ endif
+ CFLAGS += -W -Wall -Wextra -Wchar-subscripts -Wmissing-prototypes -Wmissing-declarations -Wno-switch -Wno-unused -Wredundant-decls -Wshadow
+@@ -238,34 +238,34 @@ $(OBJDIR)/__longjmp.o $(OBJDIR)/setjmp.o \
+ $(OBJDIR)/clone.o
+ 
+ $(OBJDIR)/dietlibc.a: $(DIETLIBC_OBJ) $(OBJDIR)/start.o
+-	$(CROSS)ar cru $@ $(DIETLIBC_OBJ)
++	$(CROSS)$(AR) cru $@ $(DIETLIBC_OBJ)
```

Comment:
Mh, I think that `$(CROSS)$(AR)` is wrong.

Reply:
I think you're right, we will get a 'double' cross if it's set.